module-infrastructure/persistence-db/src/main/kotlin/site/yourevents/guest/repository/GuestJPARepository.kt

```diff
@@ -40,4 +40,7 @@ interface GuestJPARepository : JpaRepository<GuestEntity, UUID> {
                 "AND g.attendance = false " +
                 "AND g.invitation.deleted = false")
     fun findNotAttendGuestsByInvitation(invitationEntity: InvitationEntity): List<GuestEntity>
+
+    @Query("SELECT g FROM guest g WHERE g.member.id = :memberId AND g.invitation.id = :invitationId")
```

Comment:
```suggestion
    @Query("SELECT g.attendance FROM guest g WHERE g.member.id = :memberId AND g.invitation.id = :invitationId")
```

attendance 값만 가져온다면, 굳이 Guest 객체 전체를 반환하기보다는 필요한 필드만 반환하는 것이 좋을 것 같습니다. 👍

Reply:
아 그렇군요!! 너무 많은걸 반환하지 않는 것이 더욱 좋아보입니다!! 수정하였습니다.

Reply:
저희가 필요한 값이 `attendance` 값인데, 해당 메서드에서 Guest 객체를 가져오고, 서비스 로직에서 아래와 같이 참조가 일어나고 있네요, 이 부분을 수정하게 된다면, 승현님께서 말씀하신 예외도 발생하지 않고, 성능 향상에 도움이 될 것 같습니다!
```kotlin
return guest.attendance
```

Reply:
반환 타입을 `GuestEntity`가 아니라 `Boolean` 값으로 수정하면 예찬님이 말씀하신 대로 우려하신 부분이 일어나지는 않을 것 같습니다..!

Reply:
아 그렇군요!! 쿼리문만 수정하는 것을 생각했지 반환타입의 수정을 생각하지 못했던 것 같습니다!!! 수정하겠습니다!!